bigtable-hbase-1.x-parent/bigtable-hbase-1.x-mapreduce/src/main/java/com/google/cloud/bigtable/mapreduce/hbasesnapshots/ImportHBaseSnapshotJob.java

```diff
@@ -0,0 +1,290 @@
+    conf.setIfUnset(
+        IMPORT_SNAPSHOT_JOBNAME_KEY,
+        ImportHBaseSnapshotJob.class.getName()
+            + " - from snapshot "
+            + conf.get(SNAPSHOTNAME_KEY)
+            + " to "
+            + conf.get(TABLENAME_KEY));
+
+    // implicit bigtable configs
+    BigtableConfiguration.configure(
+        conf,
+        conf.get(BigtableOptionsFactory.PROJECT_ID_KEY),
+        conf.get(BigtableOptionsFactory.INSTANCE_ID_KEY),
+        conf.get(BigtableOptionsFactory.APP_PROFILE_ID_KEY, ""));
+    conf.setBooleanIfUnset(BigtableOptionsFactory.BIGTABLE_USE_TIMEOUTS_KEY, true);
+    conf.setIfUnset(
+        BigtableOptionsFactory.BIGTABLE_MUTATE_RPC_TIMEOUT_MS_KEY,
+        String.valueOf(BIGTABLE_MUTATE_RPC_TIMEOUT_MS_DEFAULT));
+
+    // implicit table outputformat configs that are used in the job to write map output to a table
+    conf.set(TableOutputFormat.OUTPUT_TABLE, conf.get(TABLENAME_KEY));
+    conf.setStrings(
+        "io.serializations",
+        conf.get("io.serializations"),
+        MutationSerialization.class.getName(),
+        ResultSerialization.class.getName());
+
+    return 0;
+  }
+
+  /**
+   * Sets up the actual job.
+   *
+   * @param conf The current configuration.
+   * @return The newly created job.
+   * @throws java.io.IOException When setting up the job fails.
+   */
+  protected static Job createSubmittableJob(Configuration conf) throws IOException {
+    Job job = Job.getInstance(conf, conf.get(IMPORT_SNAPSHOT_JOBNAME_KEY));
+    job.setJarByClass(ImportHBaseSnapshotJob.class);
+    TableMapReduceUtil.initTableSnapshotMapperJob(
+        conf.get(SNAPSHOTNAME_KEY),
+        new Scan().setMaxVersions(),
+        SnapshotMapper.class,
+        ImmutableBytesWritable.class,
+        Writable.class,
+        job,
+        true,
+        new Path(conf.get(SNAPSHOT_RESTOREDIR_KEY)));
+
+    job.setNumReduceTasks(0);
+    job.setOutputFormatClass(TableOutputFormat.class);
+
+    return job;
+  }
+
+  protected enum ScanCounter {
+    NUM_ROWS,
+    NUM_CELLS,
+  }
+
+  /** MAPPER */
+  protected static class SnapshotMapper extends TableMapper<ImmutableBytesWritable, Put> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SnapshotMapper.class);
+
+    private Boolean isEmptyRowWarned = false;
+
+    protected static final int MAX_CELLS = 100_000 - 1;
+
+    @Override
+    protected void setup(Context ctx) {
+      // noop
+    }
+
+    @Override
+    protected void map(ImmutableBytesWritable key, Result result, Context ctx)
+        throws IOException, InterruptedException {
+
+      List<Cell> cells = checkEmptyRow(result);
+      if (cells.isEmpty()) {
+        return;
+      }
+
+      ctx.getCounter(ScanCounter.NUM_ROWS).increment(1);
+      ctx.getCounter(ScanCounter.NUM_CELLS).increment(cells.size());
+
+      // Split the row into multiple puts if it exceeds the maximum mutation limit
+      Iterator<Cell> cellIt = cells.iterator();
+
+      while (cellIt.hasNext()) {
```

Comment:
I dont have the history as to why this was necessary, but the Import for SequenceFiles seems to be doing a lot more work in convertings Results to Puts. Can you take a look and comment as to the difference?

Reply:
Sure, the main differences between this & HBaseResultToMutationFn for SequenceFile Beam Import are:
* handling deletes - this isn't required as scan does read through deleted cells/rows. deletes are only returned when ["raw" mode for Scan is enabled](https://hbase.apache.org/1.4/apidocs/org/apache/hadoop/hbase/client/Scan.html#setRaw(boolean)). 
* splitting 100k - implemented to avoid [100k mutation limit](https://cloud.google.com/bigtable/docs/reference/data/rpc/google.bigtable.v2#mutaterowrequest)

lmk if there are other inputs needed here.

Reply:
can you just add a code comment with this info?

Reply:
added code comment